development/fitness-functions/rules/prevent-deprecated-imports.ts

```diff
@@ -0,0 +1,66 @@
+import { filterDiffLineAdditions } from '../common/shared';
+
+// Note: Remove this file once we have migrated all deprecated components to the design system
+
+const DEPRECATED_COMPONENT_PATHS = [
+  // Deprecated component-library paths
+  'component-library/avatar-base',
+  'component-library/avatar-favicon',
+  'component-library/avatar-icon',
+  'component-library/avatar-network',
+  'component-library/avatar-token',
+  'component-library/box',
+  'component-library/button',
+  'component-library/button-base',
+  'component-library/button-icon',
+  'component-library/button-link',
+  'component-library/button-primary',
+  'component-library/checkbox',
+  'component-library/form-text-field/deprecated',
+  'component-library/icon',
+  'component-library/modal-content/deprecated/modal-content',
+  'component-library/modal-header/deprecated/modal-header',
+  'component-library/text',
+  'component-library/text-field-search/deprecated',
```

Comment:
We have a few nested deprecated components that are still JS components but weren't replaced these could likely be replaced pretty easily. I'll look into it